java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthViewVector.java

```diff
@@ -0,0 +1,1402 @@
+  }
+
+  /**
+   * Get the memory address of buffer that manages the validity
+   * (NULL or NON-NULL nature) of elements in the vector.
+   *
+   * @return starting address of the buffer
+   */
+  @Override
+  public long getValidityBufferAddress() {
+    return validityBuffer.memoryAddress();
+  }
+
+  /**
+   * Get the memory address of buffer that stores the data for elements
+   * in the vector.
+   *
+   * @return starting address of the buffer
+   */
+  @Override
+  public long getDataBufferAddress() {
+    return viewBuffer.memoryAddress();
+  }
+
+  /**
+   * Sets the desired value capacity for the vector. This function doesn't
+   * allocate any memory for the vector.
+   *
+   * @param valueCount desired number of elements in the vector
+   */
+  @Override
+  public void setInitialCapacity(int valueCount) {
+    final long size = (long) valueCount * ELEMENT_SIZE;
+    checkDataBufferSize(size);
+    lastValueAllocationSizeInBytes = (int) size;
+    lastValueCapacity = valueCount;
+  }
+
+  /**
+   * Sets the desired value capacity for the vector. This function doesn't
+   * allocate any memory for the vector.
+   *
+   * @param valueCount desired number of elements in the vector
+   * @param density average number of bytes per variable width view element
+   */
+  @Override
+  public void setInitialCapacity(int valueCount, double density) {
+    final long size = (long) valueCount * ELEMENT_SIZE;
+    initialDataBufferSize = (int) (valueCount * density);
+    checkDataBufferSize(size);
+    lastValueAllocationSizeInBytes = (int) size;
+    lastValueCapacity = valueCount;
+  }
+
+  /**
+   * Get the density of this ListVector.
+   * @return density
+   */
+  public double getDensity() {
+    if (valueCount == 0) {
+      return 0.0D;
+    }
+    final double totalListSize = getTotalValueLengthUpToIndex(valueCount);
+    return totalListSize / valueCount;
+  }
+
+  /**
+   * Get the current capacity which does not exceed either validity buffer or value buffer.
+   * Note: Here the `getValueCapacity` has a relationship with the value buffer.
+   *
+   * @return number of elements that vector can hold.
+   */
+  @Override
+  public int getValueCapacity() {
+    final int validityCapacity = getValidityBufferValueCapacity();
+    final int valueBufferCapacity = Math.max(capAtMaxInt(viewBuffer.capacity() / ELEMENT_SIZE), 0);
+    return Math.min(valueBufferCapacity, validityCapacity);
+  }
+
+  private int getValidityBufferValueCapacity() {
+    return capAtMaxInt(validityBuffer.capacity() * 8);
+  }
+
+  /**
+   * zero out the vector and the data in associated buffers.
+   */
+  public void zeroVector() {
+    initValidityBuffer();
+    viewBuffer.setZero(0, viewBuffer.capacity());
+    clearDataBuffers();
+  }
+
+  /* zero out the validity buffer */
+  private void initValidityBuffer() {
+    validityBuffer.setZero(0, validityBuffer.capacity());
+  }
+
+  /**
+   * Reset the vector to initial state. Same as {@link #zeroVector()}.
```

Comment:
it looks like it is slightly different than zeroVector?

Reply:
I used the comment from `BaseVariableWidthVector`, didn't put much thought to it at the time. 
I agree with you. Let's remove that part.